### PR TITLE
Option to overwrite only changed files

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [ ] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.4.x`, `4.0.x`. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Cherry-picks comments from #1391 

> Implements feature request #1081 : added --minimal-update option to only overwrite output files that have changed, so as to reduce triggered build steps.

Credits go to @wheezil 

Also cc @OpenAPITools/generator-core-team

### CODE REVIEW

**Comment by wheezil** [comment]
Thanks!  Sometimes I get tripped up by gitflow.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/AbstractGenerator.java [hunk 1]
```diff
@@ -36,21 +40,73 @@
...
+     */
     public File writeToFile(String filename, String contents) throws IOException {
-        LOGGER.info("writing file " + filename);
-        File output = new File(filename);
+        return writeToFile(filename, contents.getBytes(Charset.forName("UTF-8")));
+    }
 
+    /**
+     * Write bytes to a file
+     * @param filename The name of file to write
```
Comment (ackintosh) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/AbstractGenerator.java]: [nits] Please insert blank comment line between the description and the list of tags.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/AbstractGenerator.java [hunk 2]
```diff
@@ -36,21 +40,73 @@
 
 public abstract class AbstractGenerator {
     private static final Logger LOGGER = LoggerFactory.getLogger(AbstractGenerator.class);
+    
+    /**
+     * Is the minimal-file-update option enabled?
+     * @return Option value
```
Comment (ackintosh) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/AbstractGenerator.java]: [nits] Please insert blank comment line between the description and the list of tags.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/AbstractGenerator.java [hunk 3]
```diff
@@ -36,21 +40,73 @@
...
+     */
+    public abstract boolean getEnableMinimalUpdate();
 
-    @SuppressWarnings("static-method")
+    /**
+     * Write String to a file, formatting as UTF-8
+     * @param filename The name of file to write
```
Comment (ackintosh) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/AbstractGenerator.java]: [nits] Please insert blank comment line between the description and the list of tags.

**Comment by ackintosh** [comment]
I've resolved the conflicts. https://github.com/OpenAPITools/openapi-generator/pull/1397/commits/938e34a6bcf95dd5f82ac56ee926cbacc2f1fdf2 😌

**Comment by ackintosh** [comment]
This PR can be closed? (https://github.com/OpenAPITools/openapi-generator/pull/2451 has been merged)

**Pull request closed without merging**